frontend/src/pages/retailer/ModalAddRetailer.js

```diff
@@ -0,0 +1,264 @@
+import TogglePage from "../../components/TogglePage";
+import ModalEditWarehouseRetailer from "./ModalEditWarehouseRetailer";
+import ModalAddWarehouseRetailer from "./ModalAddWarehouseRetailer";
+import {FaEdit} from "react-icons/fa/index";
+import "./styles.css"
+import * as JSONwarehouses from "react-bootstrap";
+import {forEach} from "react-bootstrap/ElementChildren";
+function ModalAddRetailer(props) {
+    const currentCustomerId = localStorage.
+    getItem("currentCustomerId") != null ? localStorage.
+    getItem("currentCustomerId"): 0;
+
+    const [retailerDto, setRetailerDto] = useState({
+        fullName: '',
+        identifier: '',
+        customerId: currentCustomerId,
+        warehouses: []
+    });
+
+    const warehouse = localStorage.
+    getItem("warehouse") != null ? localStorage.
+    getItem("warehouse"): null;
+
+    const [checkBoxes, setCheckBox] = useState([]);
+    const [warehouses, setWarehouses] = useState([]);
+    const [lgShow, setLgShow] = useState(false);
+    const [editWarehouse, setEditWarehouse] = useState({
+        editShow: false,
+        warehouse: []
+    });
+
+    const addWarehouseHandler = warehouse => {
+        setWarehouses(preState => [...preState, warehouse]);
+    }
+    const editWarehouseHandler = warehouse => {
+        setWarehouses(preState =>
+         preState.map(wh => wh.identifier === warehouse.identifier ? warehouse : wh)
+        )
+    }
+    const [errors, setErrors] = useState({
+        validationErrors: [],
+        serverErrors: ''
+    });
+
+    const handleFullName = (e) => {
+        setRetailerDto(preState => ({
+            ...preState,
+           fullName: e.target.value
+        }));
+    };
+    const handleId = (e) => {
+        setRetailerDto(preState => ({
+            ...preState,
+            identifier: e.target.value
+        }));
+    };
+
+    const addRetailerHandler = (e) => {
+
+        setErrors(preState => ({
+            ...preState,
+            validationErrors: ''
+        }));
+
+            fetch(`/customers/${currentCustomerId}/retailers`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(retailerDto)
+            })
+                .then(function (response) {
+                    if (response.status !== 201) {
+                        setErrors({
+                            serverErrors: "Something go wrong, try later",
+                            validationErrors: ''
+                        });
+                    } else {
+                        setErrors(preState => ({
+                            ...preState,
+                            validationErrors: []
+                        }));
+                        props.onChange(false, retailerDto);
+                    }
+                });
+
+    };
+
+    const handleCheckedChange = (warehouseId) => {
+            checkBoxes.push(warehouseId);
```

Comment:
We need to use here the same approach: setWarehouses(preState => [...preState, warehouse]);